catalog/dags/common/extensions.py

```diff
@@ -1,12 +1,30 @@
-EXTENSIONS = {
-    "image": {"jpg", "jpeg", "png", "gif", "bmp", "webp", "tiff", "tif", "svg"},
-    "audio": {"mp3", "ogg", "wav", "aiff", "flac", "wma", "mp4", "aac", "m4a", "m4b"},
+import mimetypes
+
+from common.constants import MEDIA_TYPES as SUPPORTED_TYPES
+
+
+FILETYPE_EQUIVALENTS = {
+    # Image extensions
+    "jpeg": "jpg",
+    "tif": "tiff",
+    # Audio extensions
+    "midi": "mid",
 }
 
 
-def extract_filetype(url: str, media_type: str) -> str | None:
-    """Extract the filetype from a media url extension."""
-    possible_filetype = url.split(".")[-1]
-    if possible_filetype in EXTENSIONS.get(media_type, {}):
-        return possible_filetype
-    return None
+def extract_filetype(url: str) -> tuple[str | None, str | None]:
+    """
+    Extract the filetype from a media url extension.
+
+    Returns only if the media type guessed is a supported type.
+    """
+    if mime_type := mimetypes.guess_type(url)[0]:
```

Comment:
The [`types_map`](https://github.com/python/cpython/blob/8edfa0b0b4ae4235bb3262d952c23e7581516d4f/Lib/mimetypes.py#L432) in Python mimetypes library is missing some types that we use, including `.mid`, `.ogg` and `.ogv` types. This is also causing the CI tests to fail.
 
We can probably add new types after initializing, using `add_type` function:
 https://github.com/python/cpython/blob/8edfa0b0b4ae4235bb3262d952c23e7581516d4f/Lib/mimetypes.py#L86
 
 I think we should also add logging to identify extensions that don't return a supported filetype here to find any other extensions that are not supported but available in our data.

Reply:
It's curious. The .mid/.midi extensions are included, and the script works well on Mac but not in the container. It seems it depends on the OS's local files. 

https://github.com/python/cpython/blob/8edfa0b0b4ae4235bb3262d952c23e7581516d4f/Lib/mimetypes.py#L592-L593

In any case, it's good to have a workaround!